packages/flame/lib/src/components/component.dart

```diff
@@ -243,6 +244,28 @@ class Component {
     }
   }
 
+  Map<Type, InheritedComponent>? _inheritedComponents;
```

Comment:
This functionality should probably be defined in the `InheritedComponent` class, instead of here.

Reply:
Thanks @st-pasha for the review.

I'm struggling to see how would the component inheritance would work if the `_inheritedComponents` lived in the `InheritedComponent`. I think, moving the `_inheritedComponent` to the `InheritedComponent` would remove O(1).

Consider a tree structure where:

InheritedComponentA has ComponentA
ComponentA has ComponentB and ComponentC
ComponentC has componendD

ComponentD should have access to InheritedComponentA since it is one of its ancestors.

The `_inheritedComponents` are always inherited from the parent component, regardless if it's an `InheritedComponent` or not. If a parent is an `InheritedComponent` it gets added to the `_inheritedComponents`.

Reply:
I agree with Pasha, since this change will make the basic component more expensive.

Reply:
@spydon how would it work then? 

Also with the current implementation a component tree with no InheritedComponents are equally expensive.

Reply:
Possibly as a mixin, it feels weird having this functionality straight on the raw component at least.

Reply:
I'm struggling to see how a mixin can work here. 

Do you mean a mixin that you have to apply to all descendants? 

So if you had a ThemingComponent that is an InheritedComponent and you have a tree of depth 40, you would apply the mixin to the 40 components?

Reply:
I haven't thought through it (currently in Berlin on vacation), but I'm fairly sure we don't want this functionality directly in the core component class at least.

Reply:
@flame-admin

Reply:
I'm looking forward to a solution where the search is O(1) without having the functionality living in the core component.

Reply:
I'm not going to work out a solution for it, but feel free to propose other solutions.

> @spydon Enjoy the vacation! I was in Berlin some weeks ago and loved it 😍

Thanks, been great so far!